doc/source/Versioning.rst

```diff
@@ -0,0 +1,134 @@
+
+.. _versioning-code:
+
+Versioning Policy
+=================
+
+This "policy" is primarily intended to codify our historic treatment of Grackle versioning.
+**Please reach out** if you have any concerns about our current policy or suggestions about how it can be improved (especially if it affects your willingness to integrate Grackle into your simulation).
+
+We always encourage users to download the latest commit when installing Grackle.
+Our general policy to keep Grackle's main branch stable and in working order.
+
+Periodically, we "release" new versions of Grackle.
+A release serves 2 purposes:
+
+  1. It's way to designate a development milestone (whether it's new functionality, changes in the public API, bugfixes) at a snapshot in the project's development.
+  2. It's a way to announce the latest changes to the library since the last milestone.
+
+Currently, the repository tracks 2 separate version numbers: the primary version number (used by the documentation and the shared/static library) **and** the pygrackle version number.
+
+.. _primary-version-number:
+
+Primary Version Number
+----------------------
+
+The version number used in a release has the form ``<MAJOR>.<MINOR>(.<MICRO>)``, and the version number tracked in the repository at any given time follows the more general form ``<MAJOR>.<MINOR>(.<MICRO>)(.dev<DEV_NUM>)``.
+In this template ``<MAJOR>``, ``<MINOR>``, and ``<MICRO>`` correspond to major, minor, and micro version numbers (the micro version number may be omitted if it’s zero). The final section can specify a development version.
+
+Version numbers associated with a release follow the rules of `semantic versioning <https://semver.org/>`__. [#f1]_
+
+
+.. COMMENT BLOCK
+
+   If we were following the rules of semantic versioning as strictly
+   as possible, we would NEVER remove deprecated functionality except
+   when incrementing the major version number.  For example, in the
+   release of Grackle 3.0, we marked some functions as deprecated.
+   Technically, if we were adhearing to the rules of semantic
+   versioning as strictly as possible, we would have waited until
+   Grackle 4.0 to actually remove those functions.  The main reason to
+   do this is if we cared about maintaining ABI compatability between
+   minor versions (it would make the shared library numbering make
+   more sense)
+
+   With all of that said, it's still technically correct for us to say
+   that we follow the semantic versioning specification. An explicit
+   exception appears to be carved out for removing deprecated
+   functionality.
```

Comment:
I know at least `yt` removes deprecated functionality in minor releases.

Reply:
I honestly don't really think it matters that much (for us). This was mostly a note to ourselves... I can remove comment if you would prefer.

In more detail, if we were developing a library where we maintained ABI compatibility, it might then be beneficial to observe this "stricter adherence" to semantic versioning... 
- Doing this would allow us to directly append the semantic version number to the end of `libgrackle.so`. For example, release version 3.4.0 of grackle would have a library called `libgrackle.so.3.4.0`.
- Since some operating systems, like linux, only look at the first of the shared library version numbers at runtime, removing deprecated functionality between minor releases would break this scheme
- (Yes, `libtool` has its own weird versioning scheme that let's you do more sophisticated things... But, my impression is that most people just do this semantic-versioning approach)

But, this discussion is purely academic. As I've noted in the docs, the current API is incompatible with maintaining ABI-compatability. Plus, it probably isn't worth the effort to aim for ABI compatability